packages/dicomImageLoader/src/imageLoader/isColorConversionRequirementsFulfilled.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * This function checks color space conversion data requirements before
+ * apply them. This function was created to solve problems like the one
+ * discussed in here https://discourse.orthanc-server.org/t/orthanc-convert-ybr-to-rgb-but-does-not-change-metadata/3533/17
+ * In this case, Orthanc server convertes the pixel data from YBR to RGB, but maintain
+ * the photometricInterpretation dicom tag in YBR
+ * @param imageFrame
+ * @param RGBA
+ * @returns
+ */
+export default function isColorConversionRequirementsFulfilled(
+  imageFrame,
+  RGBA
+) {
+  if (imageFrame === undefined) {
+    return false;
+  }
+  const { rows, columns } = imageFrame;
+  if (imageFrame.photometricInterpretation === 'YBR_FULL_422') {
```

Comment:
You can check endsWith 422 or endsWIth 420 to support all the test values we can check.

Reply:
as we talked to Bill
the formula should be edited to 

number of ybr y ybr y ybr y elements -> 3 * ceil(cols/2) + florr(cols/2) * ceil(rows/2)
+ cols * floor(rows/2)

Reply:
> as we talked to Bill the formula should be edited to
> 
> for ybr_Full_422
> 
> ```
> ybr y ybr y ybr .... (ybr or y)
> y   y  y  y  y ........ y
> ..
> ..
> ..
> (last row)
> ybr y ybr y ybr y ... (ybr or y)
> OR 
> y   y  y   y  y   y 
> ```
> 

> -> 3 * ceil(cols/2) + florr(cols/2) * ceil(rows/2) + cols * floor(rows/2)

That is:
(3 * ceil(cols/2) + floor(cols/2)) * ceil(rows/2) + cols * floor(rows/2)

Note the parenthesis around the first expression.  That breaks down into:
There are ceil(rows/2) YBR containing rows (think about 1,2,3 rows, having 1 1 and 2 YBR containing rows)
There are floor(rows/2) NON YBR containing rows.  Those have length cols as they are just y containing.
For each YBR row, there are ceil(cols/2) YBR elements - think about hte same thing as the rows, but for columns - the first, third, fifth etc column has YBR.  Each YBR has 3 elements (Y b and r)
Then, for each YBR row, there are floor(cols/2) Y element only.